src/openms/include/OpenMS/SYSTEM/File.h

```diff
@@ -55,6 +55,34 @@ namespace OpenMS
 
     friend class TOPPBase;
 
+    /**
+      @brief Class representing a temporary directory
+    
+    */
+
+    class OPENMS_DLLAPI TempDir
+    {
+    public:
+      
+      /// Construct temporary folder
+      /// If keep_dir is set to true, the folder will not be deleted on destruction of the object.
+      TempDir(bool keep_dir = false);
+
+      /// Destroy temporary folder (can be prohibited in Constructor)
+      ~TempDir();
+
+      /// Return path to temporary folder
+      const String& getPath() const;
+
+    private:
+      /// Make class noncopyable
+      TempDir(const TempDir&);
```

Comment:
while this is technically correct, it might still be confusing, because the move-assignment-op and move-ctor and not mentioned here (So one might wonder if moving is allowed and what strange things would happen if it were).
Now, since you declared a custom copy-ctor, no move* will be generated by the compiler implicitly, which is good. But to be more explicit, it would be nice to state this, and do it the C++11 way:
```
public:
   /// delete all means to copy or move a TempDir
   TempDir(const TempDir&) = delete;
   TempDir& operator=(const TempDir&) = delete;
   TempDir(TempDir&& ) = delete;
   TempDir& operator=(TempDir&&) = delete;
```